controllers/overcloudipset_controller.go

```diff
@@ -85,48 +86,58 @@ func (r *OvercloudIPSetReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 		return ctrl.Result{}, err
 	}
 
-	if instance.Status.IPAddresses == nil {
-		instance.Status.IPAddresses = make(map[string]string)
+	if instance.Status.HostIPs == nil {
+		//instance.Status.IPAddresses = make(map[string]string)
+		instance.Status.HostIPs = make(map[string]ospdirectorv1beta1.OvercloudIPSetStatus)
 	}
 
 	ctlplaneCidr := ""
 
-	// iterate over the requested Networks
-	for _, netName := range instance.Spec.Networks {
+	// iterate over the requested hostCount
+	for count := 1; count <= instance.Spec.HostCount; count++ {
+		hostname := fmt.Sprintf("%s%d", instance.Name, count)
 
-		if instance.Status.IPAddresses[netName] != "" {
-			continue
-		}
+		// iterate over the requested Networks
+		for _, netName := range instance.Spec.Networks {
 
-		network := &ospdirectorv1beta1.OvercloudNet{}
-		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: netName, Namespace: instance.Namespace}, network)
-		if err != nil {
-			if k8s_errors.IsNotFound(err) {
-				return ctrl.Result{}, nil
+			if instance.Status.HostIPs[hostname].IPAddresses[netName] != "" {
+				continue
 			}
-			// Error reading the object - requeue the request.
-			return ctrl.Result{}, err
-		}
 
-		_, cidr, _ := net.ParseCIDR(network.Spec.Cidr)
-		if network.Name == "ctlplane" {
-			ctlplaneCidr = network.Spec.Cidr
-		}
-		start := net.ParseIP(network.Spec.AllocationStart)
-		end := net.ParseIP(network.Spec.AllocationEnd)
+			network := &ospdirectorv1beta1.OvercloudNet{}
+			err := r.Client.Get(context.TODO(), types.NamespacedName{Name: netName, Namespace: instance.Namespace}, network)
+			if err != nil {
+				if k8s_errors.IsNotFound(err) {
```

Comment:
Should we log something here to inform the user that the required network is missing?  I admit, though, that I'm not sure who will be responsible for creating the networks in the first place.  Is it the user, or perhaps another controller?  Dev-tools for now?

Reply:
ack. lets log it.

Reply:
got it